### PR TITLE
#180 Fix compatibility with Bitbucket 8.0

### PR DESCRIPTION
**Pull request by mgoyal2-atl**
* Removed the usage of TaskEvent and related classes which are not available in Bitbucket 8.0.
* Using Comment events with `CommentSeverity#BLOCKER` to perform PR task related operations.
* Fixed unit tests
* Replaced `7.6.0` with `8.0.0-breakit-m13` to be used for running integration tests.
* Updated README for 7.21 support.

### CODE REVIEW

File: bitbucket-slack-server-integration-plugin/pom.xml [hunk 1]
```diff
@@ -407,7 +407,7 @@
                     <skipAllPrompts>true</skipAllPrompts>
 
                     <!-- It allows to connect to Bitbucket embedded database while it is running -->
-                    <jvmArgs>-Djdbc.url=jdbc:h2:${project.build.directory}/bitbucket/home/shared/data/db;MVCC=TRUE;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=FALSE;TRACE_LEVEL_FILE=4;AUTO_SERVER=TRUE</jvmArgs>
+                    <jvmArgs>-Djdbc.url=jdbc:h2:${project.build.directory}/bitbucket/home/shared/data/db;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=FALSE;AUTO_SERVER=TRUE</jvmArgs>
```
Comment (mgoyal2-atl) [inline bitbucket-slack-server-integration-plugin/pom.xml:410]: This is required as Bitbucket 8.0 uses H2 1.4.200 which doesn't allow MVCC setting

**Comment by mgoyal2-atl** [comment]
https://github.com/atlassian-labs/atlassian-slack-integration-server/issues/180

**Comment by mgoyal2-atl** [comment]
@mvlasovatl @utluiz Can you please add yourselves as reviewers in this PR. I am not able to add reviewers, seems like some permission issue.

**Comment by bplump** [comment]
Hi folks, we're aiming to update Atlassian's internal Bitbucket DC instance on 21 March and need an 8.0-compatible version of this plugin before we can do that. Would you be able to review this ASAP so we can get a new release and prepare for the upgrade? Thank you!

File: bitbucket-slack-server-integration-plugin/pom.xml [hunk 2]
```diff
@@ -5,7 +5,7 @@
     <groupId>com.atlassian.bitbucket.plugins</groupId>
     <artifactId>bitbucket-slack-server-integration-plugin</artifactId>
     <packaging>atlassian-plugin</packaging>
-    <version>3.0.5-SNAPSHOT</version>
+    <version>3.1.0-SNAPSHOT</version>
```
Comment (mvlasovatl) [inline bitbucket-slack-server-integration-plugin/pom.xml:8]: Minor version upgrade is usually performed during the release process and not as a PR change. Additionally, minor version is changed when some new features are added. I don't see any new features introduced in this PR. Could you revert this change?

File: bitbucket-slack-server-integration-plugin/src/main/java/com/atlassian/bitbucket/plugins/slack/listener/BitbucketNotificationEventListener.java [hunk 3]
```diff
@@ -47,48 +45,28 @@
...
-
     @EventListener
     public void onEvent(final PullRequestEvent event) {
+        if (event instanceof PullRequestCommentEvent) {
+            PullRequestCommentEvent commentEvent = (PullRequestCommentEvent) event;
+            if (commentEvent.getComment().getSeverity() == BLOCKER) { //BLOCKER severity represents a PR task
+                onEvent(commentEvent);
+                return;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (bplump) [reply bitbucket-slack-server-integration-plugin/src/main/java/com/atlassian/bitbucket/plugins/slack/listener/BitbucketNotificationEventListener.java:69]: Not really. The concept of blocker comments were introduced to the backend in 6.7 but they weren't shown as tasks until 7.0. So I suspect this code compiles and runs fine but doesn't behave as intended. 

As a result, this change means the plugin really only supports 7.0+ now.

File: .github/workflows/all-tests.yml [hunk 4]
```diff
@@ -162,7 +162,7 @@ jobs:
     strategy:
       matrix:
         java-version: [8, 11]
-        bitbucket-version: [6.7.1, 7.6.0]
+        bitbucket-version: [6.7.1, 8.0.0-breakit-m13]
```
Comment (bplump) [inline .github/workflows/all-tests.yml:165]: @mgoyal2-atl - I don't believe this code will actually work as expected on `6.7.1`. From [the API changelog for 7.2](https://developer.atlassian.com/server/bitbucket/reference/api-changelog/#tasks):

> `BLOCKER` comments have existed since 6.7, however, _prior to 7.0 they are not treated as tasks._ Apps that wish to retain semantic compatibility with 6.x must continue using the `TaskService` until support for pre-7.0 releases can be dropped.

As a result, maybe the earliest version should be 7.0 or (since that's out of support) 7.6.0.

File: bitbucket-slack-server-integration-plugin/pom.xml (hunk 2 above)
Comment (bplump) [reply bitbucket-slack-server-integration-plugin/pom.xml:8]: If that's how you folks work, fine. But Manish is doing the right thing here. 

This pull request is introducing changes that are incompatible with versions of BbS that this plugin currently supports (no new features, but still incompatible changes). Semver says he should, at least, increment the minor version number.

In addition, he's making the version change in this pull request so the merge commit that makes the incompatible changes also contains a version bump indicating that they're incompatible changes... it's a bit like self-documenting code. Otherwise, there's a chance that someone thinks "the new version is incompatible with my version of Bitbucket - I'll branch from before the version was changed to pick up compatible changes" but they'll end up getting these changes anyway.

I understand that it's subtle, but it's definitely how/when this kind of version bump should be done.

**Review by bplump — APPROVED** [review]

File: bitbucket-slack-server-integration-plugin/src/main/java/com/atlassian/bitbucket/plugins/slack/listener/BitbucketNotificationEventListener.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mgoyal2-atl) [reply bitbucket-slack-server-integration-plugin/src/main/java/com/atlassian/bitbucket/plugins/slack/listener/BitbucketNotificationEventListener.java:69]: The integration tests pass on 6.7.1, there may be some gaps in the ITs which didn't catch the potential bug. Let me try testing it manually and if it doesn't work I'll update the minimum supported version to 7.0

[the remark this answers is not in the document]
Comment (mgoyal2-atl) [reply bitbucket-slack-server-integration-plugin/src/main/java/com/atlassian/bitbucket/plugins/slack/listener/BitbucketNotificationEventListener.java:69]: I rebased my local branch onto `dev` and found out that the minimum supported version is already updated to 7.0.0 on `dev` branch in [this commit](https://github.com/atlassian-labs/atlassian-slack-integration-server/commit/6b69f2a41468067c79e26b289d5e3495e68b6812). I'll test my changes on 7.0 now.

[the remark this answers is not in the document]
Comment (mgoyal2-atl) [reply bitbucket-slack-server-integration-plugin/src/main/java/com/atlassian/bitbucket/plugins/slack/listener/BitbucketNotificationEventListener.java:69]: Tested my changes on 7.0 and they are working as expected. The PR is ready for review again

CC @mvlasovatl @bplump

[the remark this answers is not in the document]
Comment (bplump) [reply bitbucket-slack-server-integration-plugin/src/main/java/com/atlassian/bitbucket/plugins/slack/listener/BitbucketNotificationEventListener.java:69]: LGTM

I'm not too surprised that the func tests pass. They must be using REST (which will expose the concept of a `BLOCKER` comments from 6.7) or else we'd need a whole different set of page objects to support the new PR experience introduced in 7.0.

File: bitbucket-slack-server-integration-plugin/pom.xml (hunk 2 above)
Comment (mgoyal2-atl) [reply bitbucket-slack-server-integration-plugin/pom.xml]: Reverted the minor version bump

Comment (utluiz) [reply bitbucket-slack-server-integration-plugin/pom.xml]: We don't have a strict process on this because SEMVER is effectively meaningless for plugins since they are not published in any repository but manually market compatible in the Atlassian Marketplace.

Having said that, I think it would make sense bumping the version here. During the release process, we would just leave the release version as is instead of overriding it. Or maybe I'm unaware about something specific about our GitHub setup. 🤷‍♂️

**Review by utluiz — APPROVED** [review]
Changes seem good.

**Review by mvlasovatl — APPROVED** [review]
